projects/canopy/src/lib/brand-icon/brand-icon.component.ts

```diff
@@ -79,12 +93,16 @@ export class LgBrandIconComponent {
     let idCount = 0;
     let xlinkCount = 0;
 
-    return svgData
-      .replace(/id="([^"]+)"/g, () => `id="lg-brand-icon-${this.id}-${idCount++}"`)
-      .replace(
-        /xlink:href="#\w+"/g,
-        () => `xlink:href="#lg-brand-icon-${this.id}-${xlinkCount++}"`,
-      );
+    return (
+      svgData
+        // Changes the lg-icon-fill-primary id to be a class to avoid issues with duplicated ids.
+        .replace(/id="lg-icon-fill-primary"/g, () => 'data-colour="lg-icon-fill-primary"')
```

Comment:
@aaronBery using the data attribute instead of an `id` should solve the possible issue you raised :) (Andy suggested that :) )

Reply:
thanks for coming up with that solution :-)